nms/app/packages/magmalte/app/components/feg/FEGGateways.js

```diff
@@ -86,44 +84,22 @@ const useStyles = makeStyles(theme => ({
 }));
 
 function CWFGateways(props: WithAlert & {}) {
-  const [gateways, setGateways] = useState<?(federation_gateway[])>(null);
-  const {match, history, relativePath, relativeUrl} = useRouter();
-  const networkId = nullthrows(match.params.networkId);
-  const classes = useStyles();
-
-  const {isLoading} = useMagmaAPI(
-    MagmaV1API.getFegByNetworkIdGateways,
-    {networkId},
-    useCallback(
-      response =>
-        setGateways(
-          Object.keys(response)
-            .map(k => response[k])
-            .filter(g => g.id),
-        ),
-      [],
-    ),
+  const ctx = useContext(FEGGatewayContext);
+  const [gateways, setGateways] = useState<federation_gateway[]>(
+    Object.keys(ctx.state).map(gatewayId => ctx.state[gatewayId]),
```

Comment:
Used Object.keys + map to get an array of the values here instead of Object.values because Object.values returns an array of mixed type which gives a flow error. https://stackoverflow.com/questions/47187242/flow-object-type-incompatible-with-arraymixed